### ISSUE TITLE
28-1 beta - no content displayed.  eror in logs

### ISSUE THREAD

**Issue by pcause** [zadam/trilium#355]
I installed 28-1 beta and when starting get no content and a message about failed consistency check.  here is portion of log

07:39:51.446 INFO  Content hash computation took 54ms
07:39:51.446 INFO  Content hash checks PASSED
07:40:00.284 INFO  ERROR: JS Error: Uncaught error: Message: Uncaught TypeError: Cannot read property 'data' of undefined - URL: http://localhost:52731/#root/nYWXh5qiVskx/sXPBnSKO6agA/0b87RxQ3ACN1/fbdqQxfTxLVL - Line: 5 - Column: 426479 - Error object: {}
07:40:01.116 INFO  PUT /api/notes/fbdqQxfTxLVL
07:40:01.200 INFO  ERROR: JS Error: Uncaught error: Message: Uncaught TypeError: Cannot read property 'data' of undefined - URL: http://localhost:52731/#root/nYWXh5qiVskx/sXPBnSKO6agA/0b87RxQ3ACN1/fbdqQxfTxLVL - Line: 5 - Column: 426479 - Error object: {}
07:40:06.414 INFO  ERROR: JS Error: Uncaught error: Message: Uncaught TypeError: Cannot read property 'data' of undefined - URL: http://localhost:52731/#root/nYWXh5qiVskx/sXPBnSKO6agA/0b87RxQ3ACN1/fbdqQxfTxLVL - Line: 5 - Column: 426479 - Error object: {}
07:40:10.107 INFO  PUT /api/notes/fbdqQxfTxLVL
07:40:24.491 INFO  DELETE /api/branches/3nEb0UHIpJx4
07:40:24.576 INFO  GET /api/tree
07:40:24.577 INFO  Sending message to all clients: {"type":"refresh-tree"}
07:40:24.636 INFO  GET /api/tree

**Comment by pcause** [comment]
tried to use 28-0 beta again and same thing.

**Comment by zadam** [comment]
Can you please attach also the part of the logs about consistency check fail? Thanks!

**Comment by zadam** [comment]
@dixonge those are not trilium's logs. Could you please look into /home/[user]/.local/share/trilium-data/log/? Thanks!

**Comment by dixonge** [comment]
`15:15:20.904 INFO  Generated sourceId=Ax4EpJ1Tcg65 
15:15:20.904 INFO  App HTTP server starting up at port 35315 
15:15:20.905 INFO  {
  "appVersion": "0.28.0-beta",
  "dbVersion": 123,
  "syncVersion": 4,
  "buildDate": "2019-01-14T23:51:55+01:00",
  "buildRevision": "e50f9cd0a354e29bb40c161cf7288e13b732e0d3",
  "dataDirectory": "/home/donblanco/.local/share/trilium-data"
} 
15:15:21.032 INFO  GET / 
15:15:21.048 INFO  DB ready. 
15:15:21.058 INFO  Generated sourceId=zo1l47amfFXv 
15:15:21.559 INFO  GET /api/app-info 
15:15:21.562 INFO  GET /api/tree 
15:15:21.563 INFO  GET /api/script/startup 
15:15:21.583 INFO  POST /api/tree/load 
15:15:21.593 INFO  GET /api/options 
15:15:21.600 INFO  POST /api/tree/load 
15:15:21.624 INFO  GET /api/notes/1mPR6aA5LpwJ 
15:15:21.627 INFO  GET /api/notes/1mPR6aA5LpwJ/attributes 
15:15:21.640 INFO  GET /api/script/bundle/HifmpEtWKldT 
15:15:21.683 INFO  POST /api/script/exec 
15:15:21.731 INFO  GET /api/script/relation/1mPR6aA5LpwJ/runOnNoteView 
15:15:23.124 INFO  PUT /api/recent-notes/OfVBbAfslBrT/root%2F9IMeV5cfRAv9%2F1mPR6aA5LpwJ 
15:15:24.351 INFO  GET /api/notes/9IMeV5cfRAv9 
15:15:24.361 INFO  GET /api/notes/9IMeV5cfRAv9/attributes 
15:15:24.625 INFO  GET /api/script/relation/9IMeV5cfRAv9/runOnNoteView 
15:15:25.845 INFO  PUT /api/recent-notes/gVG1SINmYyDC/root%2F9IMeV5cfRAv9 
15:15:27.665 INFO  GET /api/notes/4LWVRsa1W3cs 
15:15:27.675 INFO  GET /api/notes/4LWVRsa1W3cs/attributes 
15:15:27.715 INFO  GET /api/script/relation/4LWVRsa1W3cs/runOnNoteView 
15:15:27.721 INFO  POST /api/tree/load 
15:15:29.162 INFO  PUT /api/recent-notes/3O4fgh5bW4g5/root%2F9IMeV5cfRAv9%2F4LWVRsa1W3cs 
15:15:29.516 INFO  GET /api/notes/qRoWNoUOlINF 
15:15:29.518 INFO  GET /api/notes/qRoWNoUOlINF/attributes 
15:15:29.569 INFO  GET /api/images/C2HO2CRCqnpu/trilium-small.png 
15:15:29.622 INFO  GET /api/script/relation/qRoWNoUOlINF/runOnNoteView 
15:15:29.627 INFO  POST /api/tree/load 
15:15:31.015 INFO  PUT /api/recent-notes/VoWr2aWsr314/root%2FqRoWNoUOlINF 
15:15:31.053 INFO  ERROR: Branch is not deleted even though main note is deleted for following branch IDs: 'cRtS4fxyAzVr' 
15:15:31.084 INFO  Consistency checks failed (took 33ms) with these errors: ["Branch is not deleted even though main note is deleted for following branch IDs: 'cRtS4fxyAzVr'"] 
15:15:31.084 INFO  Sending message to all clients: {"type":"consistency-checks-failed"} 
`

**Comment by pcause** [comment]
I am sure what I posted was the trilium log.  I went back to 28.0 beta and used the backup database but have saved the one with the error.  I will try to recreate and post again tomorrow

**Comment by zadam** [comment]
Yes, your log is a trilium log but unfortunately it is too short and doesn't show the consistency check error.

My comment about "those are not trilium logs" were pointed at dixonge.

**Comment by zadam** [comment]
@dixonge thank you for the logs, they have been really helpful! The only thing I could come up with is that you possibly cloned a note after you deleted the note - the sequence could be you put the note into the clipboard, deleted this note, then pasted the note somewhere - and this created the inconsistency. Given the context in #357 I consider this plausible?

But of course this is a bug in Trilium and I fixed it in the master. If you want to fix this in your document so it stops buggin you, you can execute following command in SQL console (ALT-O):

```sql
UPDATE branches SET isDeleted = 1 WHERE branchId = 'cRtS4fxyAzVr'
```

I will probably add some database self-healing for minor issues like this so it doesn't bug users so much.

**Comment by technovangelist** [comment]
I had a similar issue. Still on just the sample database and tried deleting tasks from the task list. When I close trilium and restart, I get the following in the logs:
`20:16:08.721 INFO  POST /api/tree/load
20:16:08.727 INFO  ERROR: JS Error: Can't find note "VRrLnVKrIwSs"
20:16:18.232 INFO  ERROR: Branch is not deleted even though main note is deleted for following branch IDs: 'tZXA4PT0Sd7i', 'w9YEqtZCw93U', 'Qp0jK4YzocyS', 'fIo5ToihfNKC', 'gbbBgHzgroZU'
20:16:18.249 INFO  Consistency checks failed (took 16ms) with these errors: ["Branch is not deleted even though main note is deleted for following branch IDs: 'tZXA4PT0Sd7i', 'w9YEqtZCw93U', 'Qp0jK4YzocyS', 'fIo5ToihfNKC', 'gbbBgHzgroZU'"]
20:16:18.249 INFO  Sending message to all clients: {"type":"consistency-checks-failed"}`

so then I execute the corresponding variation to the sql statement above:
`UPDATE branches SET isDeleted = 1 WHERE branchId IN ('tZXA4PT0Sd7i', 'w9YEqtZCw93U', 'Qp0jK4YzocyS', 'fIo5ToihfNKC', 'gbbBgHzgroZU')`
then close and restart.

**Comment by pcause** [comment]
I brought up the server in a browser and am seeing an empty screen.  The logs showed the errors described above.  There doesn't seem to be an alt-o for the browser, so I deleted my server directory, started the server and synced from a client.  The client finished the sync and when I bring up the server - no content.  Both are running 28.0-1.  I looked at the server log and see this:

12:22:11.087 INFO  GET /api/sync/changed?lastSyncId=800
12:22:11.459 INFO  ERROR: Branch is not deleted even though main note is deleted for following bra
nch IDs: 'pL58i31rEzu6', '8WymNEwS0JAn'                                                           12:22:11.536 INFO  Consistency checks failed (took 83ms) with these errors: ["Branch is not deleted even though main note is deleted for following branch IDs: 'pL58i31rEzu6', '8WymNEwS0JAn'"]     12:22:11.536 INFO  Sending message to all clients: {"type":"consistency-checks-failed"}

I had had an issue with the client but used the sql command to fix it and it displays just fine.  any thoughts on what I can do on the server to fix this?  I guess I can start a client instance with the TRILIUM_DATA_DIR env variable and the pop the sql console to fix.

**Comment by pcause** [comment]
I was on android and using firefox.  no alt-o but switched to chromium and could access the sql console.  Did the magic command and ctl-r and I can see my data now

**Comment by zadam** [comment]
I'm glad it worked out for you in the end. I just released new version 0.28.2 in which I fixed the underlying issue and also included "auto fixer" which would automatically fix this in the DB. I plan more comprehensive support for automatic healing of the DB for the next relase.
